content/micropython/00.first-steps/03.runtime-package/runtime-package.md

```diff
@@ -0,0 +1,367 @@
+- [Arduino MicroPython Package Installer](https://labs.arduino.cc/en/labs/micropython-package-installer) - for installing **MicroPython packages** on an Arduino board.
+
+## Installation
+
+To use the runtime package, we will need to install it first.
+
+1. Download and install the [Arduino MicroPython Package Installer](https://labs.arduino.cc/en/labs/micropython-package-installer).
+2. Connect your board to your computer.
+3. Run the tool. In the tool, you should now see your board connected.
+
+   ![Board connected.](assets/connect-device.png)
+
+4. After verifying that your board is connected, click on the search field, and search for **runtime**. Install the package.
+
+   ![Install the package.](assets/install-package.png)
+
+5. When the installation is complete, we are ready to use the package.
+
+## Basic Example
+
+We will begin by one of the most known example: blinking an LED. Let's take a look at the code example below:
+
+```python
+from arduino import *
+
+led = 'LED_BUILTIN'
+def setup():
+  print('starting my program')
+
+def loop():
+  print('loopy loop')
+  digital_write(led, HIGH)
+  delay(500)
+  digital_write(led, LOW)
+  delay(500)
+
+start(setup, loop)
+```
+
+This program has two main functions: `setup()` and `loop()`. If you are unfamiliar with this concept, here's how it works:
+
+- `setup()` - this function will run just once, at the start of a program. Like in this example, we use `print('starting my program')`.
+- `loop()` - this function will continue to run, until you disrupt the program by disconnecting the board or stopping the script.
+
+Inside of the functions, you can see that we are using `digital_write(led, HIGH)`. This is a function that will enable a pin on the board, and write it high (or low). Since we configured it at the top as `'LED_BUILTIN'`, we will control that LED on the board.
+
+At the bottom of the program, we have something called `start()`. This function will launch the program and concurrently run the `loop()` function.
+
+## Common Examples
+
+Arduino Runtime was created to simplify the code creation when programming in MicroPython, providing a more user-friendly syntax that allows you to understand the programs you create a bit better.
+
+Now that we have everything installed, and our basic example tested out, let's take a look at some of the more common examples.
+
+***The API is listed at [the end of this article](#runtime-api). You can also view the [source code on GitHub](https://github.com/arduino/arduino-runtime-mpy/tree/main) for further understanding.***
+
+### Pin Mode
+
+- `pin_mode(pin, mode)`
+
+Configures a pin as an input or an output.
+
+```python
+pin = "D6"
+
+def setup():
+  pin_mode(pin, OUTPUT)
+```
+
+### Analog Read
+
+- `analog_read(pin)`
+
+Analog read is a classic example where you read the voltage from an analog pin.
+
+```python
+pin = "A0"
```

Comment:
Definitions such as `A0`, `D1` etc are only available to our boards because we customise the build. let's make a not of this.